pkg/imgpkg/bundle/bundle.go

```diff
@@ -74,8 +79,13 @@ func (o *Bundle) pull(baseOutputPath string, ui goui.UI, pullNestedBundles bool,
 		return err
 	}
 
+	localizedImagesLockToRootRepo, notLocalizedToRootBundle, err := NewImagesLock(imagesLock, o.imgRetriever, rootBundleRepo).LocalizeImagesLock()
```

Comment:
picking conversation from our previous thread...

i think we should be localizing against the localized-repo location (not root location) since each bundle is "independent". (in our particular case both of them would be the same since copy cmd is localized everything into one repo, but that's really a decision of copy to make). so since we are now using `localizedImagesLockToRootRepo.Images` to loop over images and use that to make NewBundle with correct bundle location, we can use o.Repo() in NewImagesLock. this should also eliminate rootBundleRepo arg.

Reply:
Ok. I've made the change to localize against the 'parent' bundle instead of the 'root' bundle.